### PR TITLE
Refactor Neo4j graph into its own class

### PR DESCRIPTION
**Pull request by Mats-SX**
- Remove LoadingContext
- Extract utility functions to object helper

### CODE REVIEW

File: caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+      val values = header.slots.map { slot =>
+        slot.content.key match {
+          case Property(_, PropertyKey(keyName)) =>
+            val propValue = props.get(keyName).orNull
+            val dataType = schema(SparkColumnName.of(slot)).dataType
+            importedToSparkEncodedCypherValue(dataType, propValue)
+          case HasLabel(_, label) =>
+            labels(label.name)
+          case _: Var =>
```
Comment (s1ck) [inline caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala]: maybe align with the code style in relToRow (new line after case)

File: caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            importedRel.endNodeId()
+
+          case _: OfType =>
+            relType
+
+          case _: Var =>
+            importedRel.id()
+
+          case x =>
+            Raise.invalidArgument("One of: property lookup, startNode(), endNode(), id(), variable", x.toString)
```
Comment (s1ck) [inline caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala]: `One` -> `one` to align with Exception message

File: caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            val propValue = props.get(keyName).orNull
+            val dataType = schema(SparkColumnName.of(slot)).dataType
+            importedToSparkEncodedCypherValue(dataType, propValue)
+          case HasLabel(_, label) =>
+            labels(label.name)
+          case _: Var =>
+            importedNode.id()
+
+          case _ =>
+            Raise.impossible("Nothing else should appear")
```
Comment (s1ck) [inline caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala]: Align with `_` case in `relToRow`

File: caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraphLoader.scala [hunk 4]
```diff
@@ -48,21 +36,24 @@ object Neo4jGraphLoader {
...
 
-    val nodeSchema = nodes.aggregate(Schema.empty)({
-      case (schema, node) =>
-        val labels =
-          if (node.labels().isEmpty)  // could remove this unless we rely on `NoLabel`
-            NoLabel
-          else
-            node.labels().asScala.toSet
+    val nodeSchema = nodes.aggregate(Schema.empty)(
+      {
```
Comment (s1ck) [inline caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraphLoader.scala]: is this the preferred code style? having the `{` on a separate line?

**Comment by Mats-SX** [comment]
It's funny that you react to code style. I used the `scalafmt` config file to format these. Maybe we need to configure it a bit.

File: caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala (hunk 1 above)
Comment (Mats-SX) [reply caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala]: I don't understand; this _is_ a new line after `case`.

**Comment by s1ck** [comment]
Yeah, it should at least fix the newline after `case` scenario.

File: caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraphLoader.scala (hunk 4 above)
Comment (Mats-SX) [reply caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraphLoader.scala]: It's what is currently configured in `scalafmt`. I don't think it looks awful here, but possibly better on the prev line. Or better yet as its own inner `def`.

File: caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala (hunk 3 above)
Comment (Mats-SX) [reply caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala]: You mean align the exception?

File: caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala (hunk 1 above)
Comment (Mats-SX) [reply caps-spark/src/main/scala/org/opencypher/caps/impl/spark/io/neo4j/Neo4jGraph.scala]: Oh you mean an empty line between `case`s; I get it.

**Pull request merged**